File: java/1. Two Sum/solution.java
import java.util.HashMap;


class Solution {
        public static int[] twoSum(int[] nums, int target) {

            HashMap<Integer, Integer> map = new HashMap<Integer, Integer>(); // hashmap for O(1) time retrieval each call

            for (int i = 0; i < nums.length; i++){


                if (map.containsKey(target - nums[i])){ // target - nums[i] = the remainder number for selected iteration of numbers


                    return new int [] {map.get(target - nums[i]), i}; 
                }


                map.put(nums[i], i); // add element to the map
            }

            return null;
        }
}

File: c++/1. Two Sum/solution.cpp
class Solution {
    public:
    
        vector<int> twoSum(vector<int>& nums, int target) {
            
    
            std::unordered_map<int, int> map; // map for O(1) time retrieval each call
    
            for(int i = 0; i < nums.size(); ++i){
    
                if (map.count(target - nums[i])){ // target - nums[i] = the remainder number for selected iteration of numbers
                    return {map[target -nums[i]], i};
                }
                map[nums[i]] = i; // add element to map
                
            }
            return {};
        }
    };